sound/soc/sof/compress.c

```diff
@@ -209,6 +209,9 @@ static int sof_compr_set_params(struct snd_soc_component *component,
 	pcm->params.sample_container_bytes =
 		snd_pcm_format_physical_width(SNDRV_PCM_FORMAT_S32) >> 3;
 	pcm->params.host_period_bytes = params->buffer.fragment_size;
+	pcm->params.ext_data_length = sizeof(params->codec);
+
+	memcpy((u8*)pcm->params.data, &params->codec, sizeof(params->codec));
```

Comment:
Instead of 5 times of `sizeof(params->codec)` I would introduce a variable to hold the data_size.